Start To Work On Publisher Permission Creation

GuenoleCHERUBIN opened the pull request:
If publisher permission doesn't exist create it
Otherwize update it
Created new function CreateSocialMediaPermission

GuenoleCHERUBIN committed:
Start To Work On Publisher Permission Creation

If publisher permission doesn't exist create it
Otherwize update it
Created new function CreateSocialMediaPermission

GuenoleCHERUBIN committed:
Create A New Publisher If Doesn't Exists

Function createSocialMediaPermission create a new publisher permission
based on the form ModifySocialMediaPermission
and on the user's id

EmileCalixte commented on controllers/ManagementController.php:
```diff
@@ -61,7 +61,12 @@ private function actionModifiyAccount($id)
 
         if ($formSocialMediaPermission->load($_POST) && $formSocialMediaPermission->validate()) {
-            $this->updateSocialMediaPermission($formSocialMediaPermission, $userPermission);
+            if ($userPermission == null){
```
You should use ===

EmileCalixte replied:
Also, you should add a space between ) and {

EmileCalixte commented on controllers/ManagementController.php:
```diff
@@ -140,7 +165,7 @@ private function checkIfPermissionExists($id)
         }
-        return null;
+        return $userPermission;
     }
```
This function is useless, it always returns the result of SocialMediaPermission::findOne

alexisljn commented on controllers/ManagementController.php:
```diff
@@ -140,7 +165,7 @@ private function checkIfPermissionExists($id)
             return $userPermission;
         }
-        return null;
+        return $userPermission;
```
You always returns $userPermission

GuenoleCHERUBIN committed:
Delete Useless Function

Deleted useless function after comment

GuenoleCHERUBIN committed:
Made If Condition More Accurate

Added === instead of == to have a more accurate IF condition
in actionModifiyAccount
